Assets/Mirror/Runtime/NetworkBehaviour.cs

```diff
@@ -113,27 +113,32 @@ protected void setSyncVarHookGuard(ulong dirtyBit, bool value)
         /// </summary>
         protected readonly List<SyncObject> syncObjects = new List<SyncObject>();
 
-        /// <summary>
-        /// NetworkIdentity component caching for easier access
-        /// </summary>
-        NetworkIdentity netIdentityCache;
-
         /// <summary>
         /// Returns the NetworkIdentity of this object
         /// </summary>
+        //
+        // -> cached to avoid GetComponent call each time we use isServer etc.
+        // -> with a bool to avoid 'netIdentityCache == null' check which is
+        //    extremely costly due to Unity's custom null check.
+        //    (see 4k benchmark)
+        bool netIdentityCached;
+        NetworkIdentity netIdentityCache;
         public NetworkIdentity netIdentity
         {
             get
             {
-                if (netIdentityCache == null)
+                // cache if not cached yet
```

Comment:
This is overly complicated,  can you try this instead?:

```cs
if (null == netIdentityCache)
```
or
```cs
if ( (object)netIdentityCache == null)
```

The reason this is slow is because it is using unity's == overload.  With the suggestions above,  you would be doing a reference comparison instead which should be about as fast as the boolean